ui.apps/src/main/content/jcr_root/apps/venia/clientlibs/common/js/CommerceApi.js

```diff
@@ -0,0 +1,204 @@
+
+        async _put(endpoint, params) {
+            return this._update(endpoint, params, 'PUT');
+        }
+
+        //TODO update error checking
+        async _get(endpoint) {
+            let url = `${this.rootEndpoint}${endpoint}`;
+            return this._fetch(url);
+        }
+
+        async _delete(endpoint) {
+            let url = `${this.rootEndpoint}${endpoint}`;
+            let params = {"method": "DELETE"};
+
+            return this._fetch(url, params);
+        }
+
+        /**
+         * Retrieves the cart data in JSON format
+         * @param id the cart id
+         * @returns {Promise<T>}
+         */
+        async getCart(id) {
+            return await this._get(endpoints.guestcarts.byId(id)).then(response => response.json());
+        }
+
+        /**
+         * Creates an empty shopping cart.
+         * @returns {Promise<*>}
+         */
+        async createCart() {
+            return await this._post(endpoints.guestcarts.create);
+        }
+
+        /**
+         * Updates a cart entry
+         * @param cartId
+         * @param itemId
+         * @param sku
+         * @param qty
+         * @param quoteId
+         * @returns {Promise<*>}
+         */
+        async updateCartEntry(cartId, itemId, {sku, qty, quoteId}) {
+            let url = `${endpoints.guestcarts.itemOperation(cartId, itemId)}`;
+            let params = {
+                cartItem: {
+                    sku,
+                    qty,
+                    quote_id: quoteId
+                }
+            };
+
+            let body = {body: JSON.stringify(params)};
+            return this._put(url, body)
+
+        }
+
+        /**
+         * Adds a new cart entry
+         * @param cartId
+         * @param sku
+         * @param qty
+         * @param quoteId
+         * @returns {Promise<*>}
+         */
+        async postCartEntry(cartId, {sku, qty, quoteId}) {
+            const url = `${endpoints.guestcarts.addEntry(cartId)}`;
+            const params = {
+                cartItem: {
+                    sku,
+                    qty,
+                    quote_id: quoteId
+                }
+            };
+            const body = {body: JSON.stringify(params)};
+            return await this._post(url, body);
+        }
+
+        /**
+         * Retrieves the cart totals.
+         * @param cartId
+         * @returns {Promise<T>}
+         */
+        async getTotals(cartId) {
+            return await this._get(endpoints.guestcarts.totals(cartId)).then(response => response.json());
+        }
+
+        /**
+         * Removes an item from the cart
+         * @param cartQuote
+         * @param itemId
+         * @returns {Promise<*>}
+         */
+        async removeItem(cartQuote, itemId) {
+            return await this._delete(endpoints.guestcarts.itemOperation(cartQuote, itemId));
+        }
+
+
+    }
+
+    function onDocumentReady() {
+        const endpoint = "http://localhost/magento/rest/default/V1";
```

Comment:
This should be `/magento/rest/default/V1` only, so it takes the hostname and protocol from the dispatcher.

Reply:
Right, my bad.